src/components/AddSongMenu.tsx

```diff
@@ -0,0 +1,226 @@
+  currentSong: string;
+};
+
+interface PlayListWithTracksInterface
+  extends SpotifyApi.PlaylistObjectSimplified {
+  songsOnPlayList: SpotifyApi.PlaylistTrackObject[];
+}
+
+export default function AddSongMenu({ currentSong }: AddMenuProps) {
+  const { spotifyApi } = useContext(AppContext);
+  const [playListWithTracks, setPlayListWithTracks] = useState<
+    PlayListWithTracksInterface[]
+  >([]);
+  const [state, setState] = useState({
+    bottom: false,
+  });
+  const [playListName, setPlayListName] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const showFetchingDataMessage = <CircularProgress />;
+
+  async function fetchPlayList() {
+    try {
+      setIsLoading(true);
+      const playListData = await spotifyApi.getUserPlaylists();
+      const playlistAndSongs = await Promise.all(
+        playListData.items.map(async (list) => {
+          const { items } = await spotifyApi.getPlaylistTracks(list.id);
+          const songsOnPlayList = items;
+          return { ...list, songsOnPlayList };
+        })
+      );
+
+      setPlayListWithTracks(playlistAndSongs);
+      setIsLoading(false);
+    } catch (error) {
+      handleExpiredToken(error);
+    }
+  }
+
+  const toggleDrawer = (anchor: string, open: boolean) => () => {
+    setState({ ...state, [anchor]: open });
+    if (open) {
+      fetchPlayList();
+    }
+  };
+
+  const list = (anchor: string) => {
+    return (
+      <Box
+        sx={{ width: anchor === "bottom" ? "auto" : 250 }}
+        role="presentation"
+      >
+        <List>
+          {playListWithTracks.map((list) => {
+            const included = list.songsOnPlayList.some(
+              (list) => list.track.uri === currentSong
+            );
+
+            return (
+              <ListItem
+                key={list.id}
+                disablePadding
+                onClick={() => handleListItemClick(list)}
+                onKeyDown={(event) => {
+                  if (event.key === "Enter") {
+                    handleListItemClick(list);
+                  }
+                }}
+              >
+                <ListItemButton>
+                  <ListItemAvatar>
+                    <Avatar
+                      alt={`{list.name}`}
+                      src={list.images !== null ? list.images[0].url : ""}
+                      variant="square"
+                    />
+                  </ListItemAvatar>
+
+                  <ListItemText primary={list.name} />
+                  {included && (
+                    <ListItemText primary="Song is already on list" />
+                  )}
+                </ListItemButton>
+              </ListItem>
+            );
+          })}
+        </List>
+      </Box>
+    );
+  };
+
+  async function handleListItemClick(
+    selectedPlayList: SpotifyApi.PlaylistObjectSimplified
+  ) {
+    try {
+      const song = await spotifyApi.getMyCurrentPlayingTrack();
+      const songToAdd = song?.item?.uri as string;
```

Comment:
Using `as string` here does solve your problem but not really showing your intention. I assume you don't really want to overwrite the type. `song.item?.uri` is `string | undefined` because of the "?". If you really believe you can be sure that `song.item` exists, you can use "!" instead: `song.item!.uri`. That doesn't mess with the type of `.uri` but tells TypeScript that you assume `song.item` is not null (which means `.uri` is guaranteed to be a string).

But can you be 100% sure `.item` in not null? If not, it's a good practice to implement a fallback:

```
      const songToAdd = song?.item?.uri;
      if (!songToAdd) {
        ..
      }
      // from here on `songToAdd` will be "string" automatically
```